repose-aggregator/commons/commons-configuration/src/test/resources/unmarshallerValidator/xsd/uri-stripper.xsd

```diff
@@ -65,14 +71,174 @@
+            <!--<html:p>Specifies the XPath to the link to be modified.</html:p>-->
+            <!--</xs:documentation>-->
+            <!--</xs:annotation>-->
+            <!--</xs:element>-->
+        </xs:sequence>
+
+        <!-- TODO: Fix this when adding XML support -->
+        <!--<xs:assert vc:minVersion="1.1"-->
+        <!--xpathDefaultNamespace="##targetNamespace"-->
+        <!--test=""-->
+        <!--xerces:message="Either a json element or a xml element must be defined."-->
+        <!--saxon:message="Either a json element or a xml element must be defined."/>-->
+    </xs:complexType>
+
+    <xs:complexType name="LinkPath">
+        <xs:simpleContent>
+            <xs:extension base="xs:string">
+                <xs:attribute name="token-index" type="NonNegativeInt" use="optional">
+                    <xs:annotation>
+                        <xs:documentation>
+                            <html:p>
+                                The URI Stripper filter will parse the links in the body of a response
+                                into tokens using the '/' character as a delimiter, re-insert the stripped token
+                                from the request URI, then rebuild the link. Index '0' if the first path segment.
+                                If this attribute is not present, the filter will re-insert the stripped token into
+                                the path segment it resided prior to being stripped.
+                            </html:p>
+                        </xs:documentation>
+                    </xs:annotation>
+                </xs:attribute>
+            </xs:extension>
+        </xs:simpleContent>
+    </xs:complexType>
+
+    <xs:simpleType name="HttpMethodList">
+        <xs:list itemType="HttpMethod"/>
+    </xs:simpleType>
+
+    <xs:simpleType name="HttpMethod">
+        <xs:annotation>
+            <xs:documentation>
+                <html:p>
+                    The HttpMethod simple type defines a string
+                    enumeration of HTTP method verbs as outlined in
+                    <a href="http://www.ietf.org/rfc/rfc2616.txt">RFC2616</a>
+                    section 9.
+                </html:p>
+            </xs:documentation>
+        </xs:annotation>
+
+        <xs:restriction base="xs:string">
+            <xs:enumeration value="GET"/>
+            <xs:enumeration value="DELETE"/>
+            <xs:enumeration value="POST"/>
+            <xs:enumeration value="PUT"/>
+            <xs:enumeration value="PATCH"/>
+            <xs:enumeration value="HEAD"/>
+            <xs:enumeration value="OPTIONS"/>
+            <xs:enumeration value="CONNECT"/>
+            <xs:enumeration value="TRACE"/>
+            <xs:enumeration value="ALL"/>
         </xs:restriction>
     </xs:simpleType>
 
+    <xs:simpleType name="LinkSegmentMismatchAction">
+        <xs:restriction base="xs:string">
+            <xs:enumeration value="keep"/>
+            <xs:enumeration value="remove"/>
```

Comment:
It will. Still, the JSONPath or token index could be bad, at which point we have to make a decision. This just puts the ball in the user's court.